test_conformance/printf/util_printf.cpp

```diff
@@ -362,6 +376,16 @@ std::vector<printDataGenParameters> printFloatGenParameters = {
     // xh.hhhhpAd style,default(right)-justified
 
     { { "%10.2a" }, "9990.235" },
+
+    //(Minimum)Ten-wide,two positions after the decimal,with
+    // a blank spaces inserted before the value, default(right)-justified
+
+    { { "% 10.2f" }, "1.2345" },
```

Comment:
Done. Isn't this too aggressive requirement ? This approach leads to testing very simple cases especially for small numbers. Moreover, there are already many previous cases in `util_printf.cpp` with conversion error.

Reply:
> This approach leads to testing very simple cases especially for small numbers.

If you want to change the number being tested that's fine with me, as long as it's representable in fp32.

> Moreover, there are already many previous cases in `util_printf.cpp` with conversion error.

We should probably rewrite those one day, and perhaps have a dedicated printf test for fp64 values that is only run when `cl_khr_fp64` is supported?  But that's a separate issue from this.

Reply:
I just went through #2009. It seems like `789456123.0` is rounded to float and then the result is +7.89456128E+08 - exactly the same result for C99 printf (witn `789456123.f` argument). Correct me if I am mistaken but the test assumes `cl_khr_fp64` is available. Shouldn't we distinguish float and double input arguments for printf instead of limiting input floats to "representable in fp32" ? Rounding should work exactly the same in host and device I presume.

Reply:
My understanding of the issue is as follows:

For computing the reference value on the host, `strtof` is used in `floatRefBuilder`, so the expected result is a print of `789456128.f`, regardless of whether `cl_khr_fp64` is supported or not.

For devices that do not support `cl_khr_fp64`, the OpenCL C specification (Section 6.3.1.1) says that "implementations may implicitly cast double-precision floating-point literals to single-precision literals", hence the device will print the `789456128.f` constant, matching the host.

For devices that do support `cl_khr_fp64`, the printf argument on the device will stay as `789456123.0` so the device prints the `789456123.0` constant, mismatching the host result (which used `strtof`).

> Shouldn't we distinguish float and double input arguments for printf instead of limiting input floats to "representable in fp32" ?

Yes we should distinguish float and double.  This test is called "float" so I'd argue we should only test fp32 here and we need a separate printf test for `double` (which would use `strtod` in its reference builder), just as we have a separate printf test for `half`.

Reply:
We have already such point in work plan for #1058 so I will clean this up with new PR